drivers/modem/modem_pin.c

```diff
@@ -36,12 +36,18 @@ int modem_pin_write(struct modem_context *ctx, u32_t pin, u32_t value)
 				ctx->pins[pin].pin, value);
 }
 
-int modem_pin_config(struct modem_context *ctx, u32_t pin, int flags)
+int modem_pin_config(struct modem_context *ctx, u32_t pin, bool enable)
 {
+	gpio_flags_t flags = GPIO_DISCONNECTED;
```

Comment:
`GPIO_DISCONNECTED` flag is badly supported, i.e. many GPIO drivers will return `-ENOTSUP`. This would break the modem driver on many platforms. It's better to stay with `GPIO_INPUT`. PWR_ON pin on Sara R4 module has an internal pull-up, so switching a pin to input is safe.

Reply:
OK, I'll queue this for the next miscellaneous batch.

Reply:
This change is a SQUASH commit pending in #22279.